lib/mysql/mysql.go

```diff
@@ -693,8 +715,8 @@ func (c *Connection) Connect(conn net.Conn) error {
 		c.ConnectionLog.Handshake = packet
 	case *ERRPacket:
 		c.ConnectionLog.Error = packet
-		log.Debugf("Got error packet: 0x%x / %s", p.ErrorCode, p.ErrorMessage)
-		return fmt.Errorf("Server returned error after connecting: error_code = 0x%x; error_message = %s", p.ErrorCode, p.ErrorMessage)
+		log.Debugf("Got error packet: %s", p.Error())
+		return p.GetScanError()
```

Comment:
Before this, it giving the `SCAN_UNKNOWN_ERROR` status. Returning the `ScanStatus` allows it to detect the proper `SCAN_APPLICATION_ERROR`.